src/platform/scancode.rs

```diff
@@ -0,0 +1,31 @@
+#![cfg(any(
+    target_os = "windows",
+    target_os = "macos",
+    target_os = "linux",
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd"
+))]
+
+// TODO: Maybe merge this with `modifier_supplement` if the two are indeed supported on the same
+// set of platforms
+
+use crate::keyboard::KeyCode;
+
+pub trait KeyCodeExtScancode {
```

Comment:
Could `Scancode` be renamed with `PhysicalKey` in this context?

Reply:
You mean you'd like it to be something like this?
```rust
pub trait KeyCodeExtPhysicalKey {
    fn to_physical_key(self) -> Option<u32>;
    fn from_physical_key(scancode: u32) -> KeyCode;
}
```

Reply:
Right, if we can get rid of `scancode` in favor of `physical_key` on the user facing api I think this would be very helpful

Reply:
I don't like this because if we go with this, then the term "physical_key" would be used for two different things.
1. `physical_key` in `KeyEvent` (and `RawKeyEvent`) which is of type `KeyCode` and is a platfom agnostic value.
2. `physical_key` to refer to platform native physical key identifier such as the `u16` scancodes on Windows and the `u16` `keyCode` on macOS

Reply:
An alternative name I could think of would be "native key id" as shown below. But I'm not sure what's wrong with "scancode". Is it that scancode is usualy used to refer toPS/2 scancodes, which are only used on Windows, but not on other platforms?
```rust
pub trait KeyCodeExtNativeKeyId {
    fn to_native_key_id(self) -> Option<u32>;
    fn from_native_key_id(native_key_id: u32) -> KeyCode;
}
```

Reply:
Thanks for explanation, it wasn't too clear right now the difference between physical key and scancode to me.

Reply:
> But I'm not sure what's wrong with "scancode". Is it that scancode is usualy used to refer toPS/2 scancodes, which are only used on Windows, but not on other platforms?

It was a bit difficult to grasp the differences as we introduce new semantics while also keeping existing ones. It also says in the docs that they are kinda similar. But I'm fine with the current usage as well.

> physical key (i.e. it's mostly synonymous with a scancode).

Reply:
Part of the problem is that scancode is an overloaded term. "Native key ID" seems a lot more neutral.

Reply:
I would be okay with changing this to "native key id".  I'm wondering if this should be brought up at #753 to give a chance for others to object to it.